build.gradle

```diff
@@ -52,14 +52,16 @@ apply from: "gradle/java-library.gradle"
 dependencies {
     compile 'net.bytebuddy:byte-buddy:1.6.11', 'net.bytebuddy:byte-buddy-agent:1.6.11'
 
-    provided "junit:junit:4.12", "org.hamcrest:hamcrest-core:1.3"
+    compileOnly "junit:junit:4.12", "org.hamcrest:hamcrest-core:1.3"
     compile "org.objenesis:objenesis:2.5"
 
     testCompile 'org.ow2.asm:asm:5.1'
 
     testCompile 'org.assertj:assertj-core:1.7.1'
 
-    testRuntime configurations.provided
+    //putting 'provided' dependencies on test compile and runtime classpath
+    testCompileOnly configurations.compileOnly
```

Comment:
Yup, compileOnly is nice but good old provided is a bit more intuitive to use (because does not require to set it up on testRuntime, testCompile, etc).